base/namedtuple.jl

```diff
@@ -0,0 +1,91 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+@generated function namedtuple(::Type{NamedTuple{names,T} where T}, args...) where names
```

Comment:
That would depend on having some kind of dedicated syntax thought, right? Like `((x=y for (x, y) in zip(syms, vals))...)` or curly braces or something?